Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -119,6 +119,30 @@ bool EventRegionContext::contains(const IntRect& rect) const
 
 #if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
 
+static std::optional<IntRect> guardRectForRegionBounds(const IntRect& regionBounds)
```

Comment:
thanks! I think it helps a lot to have this logic extracted here
nit: I'd group it with `shouldConsolidateInteractionRegion` and remove the last empty line

Reply:
I didn't move it down since it's a static method and needs to be implemented or declared before it's used.